terraform/jenkins/binary_cache_signing.tf

```diff
@@ -0,0 +1,47 @@
+# nix-store --generate-binary-cache-key foo secret-key public-key
+# terraform import secret_resource.binary_cache_signing_key "$(< ./secret-key)"
+resource "secret_resource" "binary_cache_signing_key" {
+  lifecycle {
+    prevent_destroy = true
```

Comment:
what is our resource lifecycle policy in general? Do we have it documented somewhere?

Reply:
This is mostly a safeguard against terraform deleting that resource from the state, as it only lives in there, in case it gets moved around and renamed etc. It is a recommended best practice as per https://github.com/numtide/terraform-provider-secret?tab=readme-ov-file#example. 

I think the rest of this question is more one about policy, not about this code particularly.